_ingest-pipelines/processors/dot-expander.md

```diff
@@ -0,0 +1,373 @@
+
+```json
+{
+  "_index": "testindex1",
+  "_id": "1",
+  "_version": 1,
+  "_seq_no": 3,
+  "_primary_term": 1,
+  "found": true,
+  "_source": {
+    "user": {
+      "address": {
+        "city": "Denver",
+        "state": "CO"
+      }
+    }
+  }
+}
+```
+
+## The `path` parameter
+
+You can use the `path` parameter to specify the path to a dotted field within an object. For example, the following pipeline specifies the `address.city` field that is located within the `user` object: 
+
+```json
+PUT /_ingest/pipeline/dot-expander-pipeline
+{
+  "description": "Dot expander processor",
+  "processors": [
+    {
+      "dot_expander": {
+        "field": "address.city",
+        "path": "user"
+      }
+    },
+    {
+      "dot_expander":{
+       "field": "address.state",
+       "path": "user"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+You can simulate the pipeline as follows: 
+
+```json
+POST _ingest/pipeline/dot-expander-pipeline/_simulate
+{
+  "docs": [
+    {
+      "_index": "testindex1",
+      "_id": "1",
+      "_source": {
+        "user": {
+          "address.city": "New York",
+          "address.state": "NY"
+        }
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+The `dot_expander` processor transforms the document into the following structure:
+
+```json
+{
+  "user": {
+    "address": {
+      "city": "New York",
+      "state": "NY"
+    }
+  }
+}
+```
+
+## Field name conflicts
+
+If a field already exists with the same path as the path to which the `dot_expander` processor should expand the value, the processor merges the two values into an array.
+
+Consider the following pipeline that expands the field `user.name`:
+
+```json
+PUT /_ingest/pipeline/dot-expander-pipeline
+{
+  "description": "Dot expander processor",
+  "processors": [
+    {
+      "dot_expander": {
+        "field": "user.name"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+You can simulate the pipeline with a document containing two values with the exact same path `user.name`:
+
+```json
+POST _ingest/pipeline/dot-expander-pipeline/_simulate
+{
+  "docs": [
+    {
+      "_index": "testindex1",
+      "_id": "1",
+      "_source": {
+        "user.name": "John", 
+        "user": {
+          "name": "Steve"
+        }
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+The response confirms that the values were merged into an array:
+
+```json
+{
+  "docs": [
+    {
+      "doc": {
+        "_index": "testindex1",
+        "_id": "1",
+        "_source": {
+          "user": {
+            "name": [
+              "Steve",
+              "John"
+            ]
+          }
+        },
+        "_ingest": {
+          "timestamp": "2024-01-17T01:44:57.420220551Z"
+        }
+      }
+    }
+  ]
+}
+```
+
+If a field contains the same name but a different path, then the field needs to be renamed. For example, the following simulate call returns a parse exception:
```

Comment:
Should "simulate" be in code font (my brain wants to incorrectly change it to "simulated")?